### DISCUSSION TITLE
Nuxt Middleware & Supabase Authentication

### DISCUSSION THREAD

**Discussion by philbaker4** [nuxt/nuxt#8688] [Questions (Nuxt 2)]
New to Nuxt  -- I'm looking to utilize [Supabase](https://supabase.io/docs/guides/auth) Auth within a Nuxt app, specifically within [middleware](https://nuxtjs.org/examples/middleware-router) for SSR routes. Session that is created from on the client when the user signs in isn't available when the middleware is running on the server. Has anyone approached this problem (or a similar problem) before? 

Here's a NextJS example, potentially helpful to see approach taken there: https://github.com/vercel/next.js/tree/canary/examples/with-supabase-auth-realtime-db

**Comment by HuxleyMc** [comment]
Hey Philbaker4

Here is an example I have started working on [NuxtJS-Supabase-auth](https://github.com/HuxleyMc/NuxtJS-Supabase-auth)

It is still a WIP and is not perfect but it does the job

**Reply by scottrobertson** [reply]
Hey @HuxleyMc 

Just an FYI, i made: 

https://github.com/scottrobertson/nuxt-supabase
https://github.com/scottrobertson/vue-supabase

So you can save writing plugins each time 👍

**Reply by codiini** [reply]
@scottrobertson does the nuxt-supabase package deal with the SSR issue?

**Comment by scottrobertson** [comment]
Hey

I started putting this together: https://github.com/nuxt-community/auth-module/issues/1082 to make it work with Nuxt Auth.

But if you just want to do it directly, just create a Middleware like. Not sure about the server side though yet.

```js
import { Middleware } from '@nuxt/types'

// The supabase client
import supabase from '~/plugins/supabase' 

const authMiddleware: Middleware = (context) => {
  if (context.route.fullPath !== '/login' && !supabase.auth.user()) {
    return context.redirect('/login')
  }
}

export default authMiddleware
```
